packages/components/src/popover/index.js

```diff
@@ -187,23 +187,34 @@ const Popover = (
 
 	/**
 	 * Offsets the position of the popover when the anchor is inside an iframe.
+	 *
+	 * Store the offset in a ref, due to constraints with floating-ui:
+	 * https://floating-ui.com/docs/react-dom#variables-inside-middleware-functions.
 	 */
-	const frameOffset = useMemo( () => {
+	const frameOffset = useRef();
+	useLayoutEffect( () => {
 		const { defaultView } = ownerDocument;
 		const { frameElement } = defaultView;
 
 		if ( ! frameElement || ownerDocument === document ) {
 			return undefined;
 		}
 
-		const iframeRect = frameElement.getBoundingClientRect();
-		return { x: iframeRect.left, y: iframeRect.top };
+		const updateFrameOffset = () => {
+			const iframeRect = frameElement.getBoundingClientRect();
+			frameOffset.current = { x: iframeRect.left, y: iframeRect.top };
+		};
+		updateFrameOffset();
+		defaultView.addEventListener( 'resize', updateFrameOffset );
```

Comment:
Wouldn't make more sense to add the `resize` ev listener in the same `useEffect` where we also add the `scroll` ev listener ?

Reply:
Sounds good. 👍 

I've pushed a commit that does that.